src/Build.UnitTests/WarningsAsMessagesAndErrors_Tests.cs

```diff
@@ -193,6 +190,125 @@ public void TreatWarningsAsMessagesWhenSpecifiedThroughAdditiveProperty()
             VerifyBuildMessageEvent(logger);
         }
 
+
+        [Fact]
+        public void TreatWarningsAsMessagesWhenSpecifiedThroughAdditivePropertyNoPrefix()
+        {
+            MockLogger logger = ObjectModelHelpers.BuildProjectExpectSuccess(
+                GetTestProject(
+                    customProperties: new List<KeyValuePair<string, string>>
+                    {
+                        new KeyValuePair<string, string>("WarningsAsMessages", "123"),
+                        new KeyValuePair<string, string>("WarningsAsMessages", $@"$(WarningsAsMessages);
+                                                                                       {ExpectedEventCode.ToLowerInvariant()}"),
+                        new KeyValuePair<string, string>("WarningsAsMessages", "$(WarningsAsMessages);ABC")
+                    }));
```

Comment:
I'm not sure I understand these tests. How does this mechanism of setting the properties apply against raised warnings to validate that warnings are treated properly?

Reply:
Those are quite confusing - a comment with desription of what we are testing and turining all of those to a single data driven test migh possibly help understandability?